lib/openstax/path_prefixer/action_controller_base_extensions.rb

```diff
@@ -47,5 +43,23 @@ def url_options
     original_url_options.dup.tap{|options| options[:script_name] = request.script_name}
   end
 
+  def _path_prefixer_app_has_route_for_path(app:, path:)
+    return false unless app.respond_to?(:routes)
+
+    app.routes.set.any? do |route|
+      next false unless "GET".match(route.verb)
+      next false unless route.path.match(path)
+
+      # Look for engine routes
+      inner_app = route.app.app
```

Comment:
do all routes have `app.app`? will this fail if we've matched a route that isn't an engine thingy?